src/components/utils/file.js

```diff
@@ -1,16 +1,106 @@
 import { findKey } from '~utils/object'
-import { isOfMimeType } from '~utils/blob'
+import { isOfMimeType, canvasToBlob } from '~utils/blob'
 
-const defaultFileTypesAccepted = ['jpg', 'jpeg', 'png', 'pdf']
-const maxFileSizeAcceptedByAPI = 10000000 // The Onfido API only accepts files below 10 MB
-export const validateFileTypeAndSize = (
+const DEFAULT_ACCEPTED_FILE_TYPES = ['jpg', 'jpeg', 'png', 'pdf']
+const MAX_FILE_SIZE_ACCEPTED_BY_API = 10000000 // The Onfido API only accepts files below 10 MB
+const MAX_IMAGE_FILE_SIZE_ACCEPTED = 3000000
+const validateFileTypeAndSize = (
   file,
-  acceptedTypes = defaultFileTypesAccepted
+  acceptedTypes = DEFAULT_ACCEPTED_FILE_TYPES
 ) =>
   findKey(
     {
       INVALID_TYPE: (file) => !isOfMimeType(acceptedTypes, file),
-      INVALID_SIZE: (file) => file.size > maxFileSizeAcceptedByAPI,
+      INVALID_IMAGE_SIZE: (file) =>
+        file.type.match(/image.*/) && file.size > MAX_IMAGE_FILE_SIZE_ACCEPTED,
+      INVALID_SIZE: (file) => file.size > MAX_FILE_SIZE_ACCEPTED_BY_API,
     },
     (checkFn) => checkFn(file)
   )
+
+const resizeImageFile = (file, onImageResize) => {
+  const reader = new FileReader()
+  reader.onload = (readerEvent) => {
+    const image = new Image()
+    image.onload = () => {
+      const resizeTo = getDimensionsToResizeTo(image)
+      const resizedWidth = resizeTo.width
+      const resizedHeight = resizeTo.height
+      const tempCanvas = document.createElement('canvas')
+      tempCanvas.width = resizedWidth
+      tempCanvas.height = resizedHeight
+      tempCanvas
+        .getContext('2d')
+        .drawImage(image, 0, 0, resizedWidth, resizedHeight)
+      const imgDiff = {
+        resizedFrom: {
+          width: image.width,
+          height: image.height,
+          fileSize: file.size,
+        },
+        resizedTo: {
+          width: resizedWidth,
+          height: resizedHeight,
+        },
+      }
+      return canvasToBlob(
+        tempCanvas,
+        (blob) => onImageResize({ resizedImage: blob, imgDiff }),
+        file.type
+      )
+    }
+    image.src = readerEvent.target.result
+  }
+  reader.readAsDataURL(file)
+}
+
+export const getDimensionsToResizeTo = (image) => {
+  // 1440px because we want to conservatively resize for Web SDK
+  // compared to mobile SDKs' 720p (1280×720px) as their UI always has a frame
+  const MAX_SIZE_IN_PIXEL = 1440
```

Comment:
shoundt these const be outside the funtion? (i don't know JS best practices, and it won't do any real difference. But "static" constants always go outside functions so they don't get assigned multiple times.

Reply:
True. I kept it here because it's only ever used in this function, but same can be said for `MAX_FILE_SIZE_ACCEPTED_BY_API` that is only ever used in `validateFileTypeAndSize`. Will move it out for consistency.